ignite/pkg/xurl/xurl.go

```diff
@@ -93,3 +122,30 @@ func IsLocalPath(address string) bool {
 func IsHTTP(address string) bool {
 	return strings.HasPrefix(address, "http")
 }
+
+func parseURL(s string) (*url.URL, error) {
+	if s == "" {
+		return nil, errors.New("url is empty")
+	}
+
+	// Handle the case where the URI is an IP:PORT or HOST:PORT
```

Comment:
Why is that the case, because I see that there is this: https://pkg.go.dev/net/url#URL.Port

Uh, what's the exact problem, can't `url` package parse addresses with ip addresses?

Reply:
I'm adding a bit more info to the source code comment so it is clear.

The issue comes from the fact that we have cases where the URL scheme is not present and in those cases the "IP:PORT" is parsed and treated as the URL path by the `url.Parse` function, leaving the colon as part of the path, which is invalid.

Reply:
Source comments added [here](https://github.com/ignite-hq/cli/pull/2392/commits/4b8cdb8df14c7b5350de81133758dd49f257a832#diff-5331d0be45b8f5101344edd2fba01b94d1eab5c1e721d4feb3987762576cbb2dR129)